### PR TITLE
ci: add license compliance workflow

### LINKED ISSUE #1005: Check license of top-level dependencies for all integrations
**Is your feature request related to a problem? Please describe.**
Similar to the license compliance workflow in Haystack core, we should check the license of top-level dependencies in haystack-core-integrations. 
Here is the workflow in Haystack: https://github.com/deepset-ai/haystack/blob/1d4883f1789dd281f0a3de00854ec547bfb19213/.github/workflows/license_compliance.yml#L49
The result of the top-level dependency check could be shown for each integration in the [inventory table](https://github.com/deepset-ai/haystack-core-integrations/tree/main?tab=readme-ov-file#inventory) in the readme of the repo.

**Describe the solution you'd like**
* Add a license check workflow that is similar to the one we use in Haystack.
* Merging all dependencies into one long list and doing only a single check instead of one check per integration

**Describe alternatives you've considered**
Not adding the checks, which poses legal risks and adoption risks.

**Additional context**
We should check with @ArzelaAscoIi if he can help. We should also check how to connect this with FOSSA. It's different than Haystack core because haystack-core-integrations has the code for multiple pypi packages in one repo.

### PR DESCRIPTION
**Pull request by anakin87**
### Related Issues

- fixes #1005

### Proposed Changes:
Add a single workflow that checks the license compliance on PRs and nightly

Mostly based on https://github.com/deepset-ai/haystack/blob/main/.github/workflows/license_compliance.yml

### How did you test it?
I tested it using this branch
test on PRs: https://github.com/deepset-ai/haystack-core-integrations/actions/runs/10563356632/job/29263394604?pr=1020
test nightly: https://github.com/deepset-ai/haystack-core-integrations/actions/runs/10563449108/job/29263683646?pr=1020

### Checklist

- I have read the [contributors guidelines](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CODE_OF_CONDUCT.md)
- I have updated the related issue with new insights and changes
- I added unit tests and updated the docstrings
- I've used one of the [conventional commit types](https://www.conventionalcommits.org/en/v1.0.0/) for my PR title: `fix:`, `feat:`, `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`.

### CODE REVIEW

File: .github/workflows/CI_license_compliance.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  EXCLUDE_PACKAGES: "(?i)^(deepeval|cohere|fastembed|ragas|haystack-pydoc-tools|tqdm|psycopg).*"
+
+  # Exclusions must be explicitly motivated
+  #
+  # - deepeval is Apache 2.0 but the license is not available on PyPI
+  # - cohere is MIT but the license is not available on PyPI
+  # - fastembed is Apache 2.0 but the license on PyPI is unclear ("Other/Proprietary License (Apache License)")
+  # - ragas is Apache 2.0 but the license is not available on PyPI
+
+  # - haystack-pydoc-tools is Apache 2.0 but the license is not available on PyPI
```
Comment (anakin87) [inline .github/workflows/CI_license_compliance.yml:24]: we can fix this

File: .github/workflows/CI_license_compliance.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  # - tqdm is MLP but there are no better alternatives
+  # - psycopg is LGPL-3.0
```
Comment (anakin87) [inline .github/workflows/CI_license_compliance.yml:26]: unsure about this. In its independent scan, FOSSA is fine with this license.

File: .github/utils/pyproject_to_requirements.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+def main(pyproject_path: Path, exclude_optional_dependencies: bool = False):
+    content = toml.load(pyproject_path)
+    deps = set(content["project"]["dependencies"])
+
+    if not exclude_optional_dependencies:
+        optional_deps = content["project"].get("optional-dependencies", {})
+        for dep_list in optional_deps.values():
+            deps.update(dep_list)
+
+    sys.stdout.write("\n".join(sorted(deps)) + "\n")
```
Comment (silvanocerza) [inline .github/utils/pyproject_to_requirements.py:15]: Was `print` not good?

File: .github/utils/pyproject_to_requirements.py
Comment (silvanocerza) [inline .github/utils/pyproject_to_requirements.py:1]: The file is missing a newline at EOF.

File: .github/workflows/CI_license_compliance.yml
Comment (silvanocerza) [inline .github/workflows/CI_license_compliance.yml:1]: Missing new line here too.

**Review by silvanocerza — APPROVED** [review]
It looks good, feel free to merge after adding the missing new lines.

File: .github/utils/pyproject_to_requirements.py
Comment (anakin87) [reply .github/utils/pyproject_to_requirements.py:1]: fixed

File: .github/workflows/CI_license_compliance.yml
Comment (anakin87) [reply .github/workflows/CI_license_compliance.yml:1]: fixed

File: .github/utils/pyproject_to_requirements.py (hunk 3 above)
Comment (anakin87) [reply .github/utils/pyproject_to_requirements.py]: copy-paste from Haystack script.
Now I'm using `print` and it works :-)

File: .github/workflows/CI_license_compliance.yml (hunk 1 above)
Comment (anakin87) [reply .github/workflows/CI_license_compliance.yml]: done.
I removed it from the exclusion, since the license is now correctly available on PyPi.

**Pull request merged**